src/napari_matplotlib/base.py

```diff
@@ -70,28 +69,43 @@ def add_single_axes(self) -> None:
         The Axes is saved on the ``.axes`` attribute for later access.
         """
         self.axes = self.figure.subplots()
-        self.apply_napari_colorscheme(self.axes)
-
-    def apply_napari_colorscheme(self, ax: Axes) -> None:
-        """Apply napari-compatible colorscheme to an Axes."""
-        # get the foreground colours from current theme
-        theme = napari.utils.theme.get_theme(self.viewer.theme, as_dict=False)
-        fg_colour = theme.foreground.as_hex()  # fg is a muted contrast to bg
-        text_colour = theme.text.as_hex()  # text is high contrast to bg
 
-        # changing color of axes background to transparent
-        ax.set_facecolor("none")
-
-        # changing colors of all axes
-        for spine in ax.spines:
-            ax.spines[spine].set_color(fg_colour)
+    def apply_style(self) -> None:
+        """
+        Any user-supplied stylesheet takes highest precidence, otherwise apply
+        the napari-compatible colorscheme (depends on the napari theme).
+        """
+        if self._apply_user_stylesheet_if_present():
+            return
+
+        stylesheet_dir = self._get_path_to_mpl_stylesheets()
+        if self.viewer.theme == "dark":
+            matplotlib.style.use(stylesheet_dir / "napari-dark.mplstyle")
+        elif self.viewer.theme == "light":
+            matplotlib.style.use(stylesheet_dir / "napari-light.mplstyle")
+        else:
+            warnings.warn(
+                f"Napari theme '{self.viewer.theme}' is not supported by"
+                " napari-matplotlib. Will fall back to the matplotlib default."
+            )
+            matplotlib.style.use("default")
```

Comment:
Here is where we're now less flexible. I figured we should fallback to the default mpl style ☝️ 

We could also do something more complicated with ``self._theme_has_light_bg`` and choosing a light or dark mpl style to contrast.